### PR TITLE
background color for paginated_data_table

### PR DESCRIPTION
**Pull request by DeTuksa**
This PR is adding an option to include background color to the PaginatedDataTable widget. By default this widget adopts the color of the card wrapping the datatable. This PR gives more flexibility to developers that want the background color of their table to be different from what they declare in their CardTheme.

1. Now the widget can either use the passed background color or the default card theme background color.
2. Flexibility for developers to decorate the datatable properly

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by DeTuksa** [comment]
This PR has passed all checks. Awaiting review

**Comment by DeTuksa** [comment]
@Hixie please review my PR

**Comment by Hixie** [comment]
@DeTuksa Reviews happen about once a week during non-holiday times but right now most of the team is not paying much attention to GitHub. I beg your patience. :-) If you haven't had a review in a week or so please reach out on the Discord (link in the contributing guidelines below).

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 1]
```diff
@@ -244,6 +245,9 @@ class PaginatedDataTable extends StatefulWidget {
   /// Defines the color of the arrow heads in the footer.
   final Color? arrowHeadColor;
 
+  ///Defines the background color of the table
```
Comment (Hixie) [inline packages/flutter/lib/src/material/paginated_data_table.dart:248]: Please see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-useless-documentation and the next section.

**Comment by goderbauer** [comment]
@DeTuksa Do you still have plans to follow up on the review comment given above? Also, as the bot pointed out above as well: This PR is missing a test. Could you please add one?

**Comment by goderbauer** [comment]
I am closing this since there has been no follow-up. Feel free to reopen when you have time to work on this again.

**Pull request closed without merging**